tools/cli/README.md

```diff
@@ -0,0 +1,108 @@
+## What
+Cadence CLI provide an command-line tool for users to perform various tasks on Cadence.  
+Users can perform operations like register, update and describe on domain;  
+also start workflow, show workflow history, signal workflow ... and many other tasks.    
+
+## How
+- Run `make bins`
+- You should see an executable `cadence`
+
+## Quick Start
+Run `./cadence` to view help message. There are some top level commands and global options.   
+Run `./cadence domain` to view help message about operations on domain  
+Run `./cadence workflow` to view help message about operations on workflow  
+(`./cadence help`, `./cadence help [domain|workflow]` will also print help messages)
+
+**Note:** make sure you have cadence server running before using CLI 
+
+### Domain operation examples 
+- Register a new domain named "samples-domain":  
+```
+./cadence --domain samples-domain domain register 
+# OR using short alias  
+./cadence --do samples-domain domain re
+```   
+- View "samples-domain" details:   
+```
+./cadence --domain samples-domain domain describe  
+```
+
+**Tips:**  
+to avoid repeated input global option **domain**, user can export domain-name in environment variable CADENCE_CLI_DOMAIN.
+```
+export CADENCE_CLI_DOMAIN=samples-domain
+
+# then just run commands without --domain flag, like
+./cadence domain desc
+```
+
+### Workflow operation examples
+(The following examples assume you already export CADENCE_CLI_DOMAIN environment variable as Tips above)
+
+- Run workflow: Start a workflow and see it's progress, this command doesn't finish until workflow completes
+```
+./cadence workflow run --tl helloWorldGroup --wt main.Workflow --et 60 -i '"cadence"'
+
+# view help messages for workflow run
+./cadence workflow run -h
+``` 
+Brief explanation:  
+To run a workflow, user must specify 
+1. Tasklist name (--tl), 
+2. Workflow type (--wt), 
+3. Execution start to close timeout in seconds (--et), 
+
+Example uses [this cadence-samples workflow](https://github.com/samarabbas/cadence-samples/blob/master/cmd/samples/recipes/helloworld/helloworld_workflow.go) 
+and it takes a string as input, so there is the `-i '"cadence"'`. Single quote `''` is used to wrap input as json. 
+
+**Note:** you need to start worker so that workflow can make progress.  
+(Run `make && ./bin/helloworld -m worker` in cadence-samples to start the worker)
+
+- Start workflow: 
+```
+./cadence workflow start --tl helloWorldGroup --wt main.Workflow --et 60 --dt 10 -i '"cadence"'
```

Comment:
I would change ./cadence to cadence in all these samples. In most cases it would be installed though brew or be somewhere in the PATH.

Reply:
For this readme, I think `./cadence` is better because reader can easily understand it is the executable after running `make`, and "advanced player" can obviously use PATH and replace it to whatever they what. 
Directly put `cadence` instead of `./cadence` will confuse beginner, if I don't have instructions on how to set PATH or use brew. And for the simplicity of this readme, I don't want to include those.